packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -197,6 +212,41 @@ class HorizontalOverflow extends Foundation<
                 itemsHeight,
             });
         }
+
+        if (this.props.onHorizontalOverflowChange) {
+            this.onHorizontalOverflowChange();
+        }
+    };
+
+    /**
+     * Handles the resize event
+     */
+    private onWindowResize = (): void => {
+        this.onHorizontalOverflowChange();
+    };
+
+    /**
+     * Checks if overflow is occuring
+     */
+    private checkOverflow(): boolean {
+        const regionWidthInformation: any = this.getItemsWidthAndTotalWidth();
+        const availableWidth: number = regionWidthInformation.availableWidth;
+        const itemWidths: number[] = regionWidthInformation.itemWidths;
+        const totalItemWidth: number = itemWidths.reduce((a: number, b: number) => a + b);
+
+        return totalItemWidth > availableWidth;
+    }
+
+    /**
+     * Callback for the horizontal overflow change
+     */
+    private onHorizontalOverflowChange = (): void => {
+        if (typeof this.props.onHorizontalOverflowChange === "function") {
+            this.props.onHorizontalOverflowChange({
+                ...this.getScrollChangeData(),
+                overflow: this.checkOverflow(),
```

Comment:
I'm actually a little lost on why we are providing a whole new callback to add a value to the scroll change object, does anyone know what the context of this decision was?

Reply:
+ @nicholasrice who may be able to keep me honest

I'll try and capture the conversation we had a couple weeks ago that led to this additional callback:
- On initialization, a user may need to know if the container has enough items to overflow (perhaps I want to hide flippers)
- A user need to know if the window has been resized and the container is now overflowing (I now want to show flippers)
- The onScrollChange event should only fire in the event of a scroll change; this is not a scroll change.
- The onHorizontalOverflowChange callback includes when there is a scroll change, but we landed that they are both useful and different enough to necessitate separate callbacks.

Reply:
+1 @janechu you were part of this discussion as well and agreed on the above points

Reply:
Sounds good, thanks for the breakdown.